app/javascript/tests/subject/CaseStatus.test.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react'
+import { shallow } from 'enzyme';
+import { Button, Modal, Form } from 'react-bootstrap';
+import CaseStatus from '../../components/subject/CaseStatus.js'
+import InfoTooltip from '../../components/util/InfoTooltip';
+import { mockPatient1, mockPatient2, mockPatient3, mockPatient4 } from '../mocks/mockPatients'
+
+const authyToken = "Q1z4yZXLdN+tZod6dBSIlMbZ3yWAUFdY44U06QWffEP76nx1WGMHIz8rYxEUZsl9sspS3ePF2ZNmSue8wFpJGg==";
+const caseStatusValues = [ '', 'Confirmed', 'Probable', 'Suspect', 'Unknown', 'Not a Case' ];
+const monitoringOptionValues = [ '', 'End Monitoring', 'Continue Monitoring in Isolation Workflow' ];
+
+function getWrapper(patient, hasGroupMembers) {
+    return shallow(<CaseStatus patient={patient} has_group_members={hasGroupMembers} authenticity_token={authyToken} />);
+}
+
+describe('CaseStatus', () => {
+    it('Properly renders all main components', () => {
+        const wrapper = getWrapper(mockPatient1, false);
+        expect(wrapper.find(Form.Label).text().includes('CASE STATUS')).toBeTruthy();
+        expect(wrapper.find(InfoTooltip).exists()).toBeTruthy();
+        expect(wrapper.find(InfoTooltip).prop('tooltipTextKey')).toEqual('caseStatus');
+        expect(wrapper.find(Form.Control).exists()).toBeTruthy();
+        expect(wrapper.find('option').length).toEqual(6);
+        caseStatusValues.forEach(function(value, index) {
+            expect(wrapper.find('option').at(index).text()).toEqual(value);
+        });
+    });
+
+    it('Changing Case Status opens modal', () => {
+        const wrapper = getWrapper(mockPatient1, false);
+        expect(wrapper.find(Modal).exists()).toBeFalsy();
+        wrapper.find(Form.Control).simulate('change', { target: { id: 'case_status', value: 'Confirmed' }, persist: jest.fn() });
+        expect(wrapper.find(Modal).exists()).toBeTruthy();
+    });
+
+    it('Properly renders modal', () => {
+        const wrapper = getWrapper(mockPatient1, false);
+        wrapper.find(Form.Control).simulate('change', { target: { id: 'case_status', value: 'Confirmed' }, persist: jest.fn() });
+        expect(wrapper.find(Modal.Title).exists()).toBeTruthy();
+        expect(wrapper.find(Modal.Title).text()).toEqual('Case Status');
+        expect(wrapper.find(Modal.Body).exists()).toBeTruthy();
+        expect(wrapper.find(Modal.Footer).exists()).toBeTruthy();
+        expect(wrapper.find(Button).at(0).text()).toEqual('Cancel');
+        expect(wrapper.find(Button).at(1).text()).toEqual('Submit');
+    });
+
+    it('Properly renders modal body and updates state when not monitoring', () => {
```

Comment:
I would change the name of this test case to be "Correctly renders modal body and does not change line list or workflow for closed record"

Reply:
done